src/diffusers/loaders.py

```diff
@@ -1333,28 +1351,39 @@ def save_function(weights, filename):
     def _convert_kohya_lora_to_diffusers(cls, state_dict):
         unet_state_dict = {}
         te_state_dict = {}
-        network_alpha = None
+        network_alphas = {}
         unloaded_keys = []
 
         for key, value in state_dict.items():
             if "hada" in key or "skip" in key:
                 unloaded_keys.append(key)
-            elif "lora_down" in key:
+            elif "lora_down" in key or "input_blocks" in key:
                 lora_name = key.split(".")[0]
                 lora_name_up = lora_name + ".lora_up.weight"
                 lora_name_alpha = lora_name + ".alpha"
                 if lora_name_alpha in state_dict:
                     alpha = state_dict[lora_name_alpha].item()
-                    if network_alpha is None:
-                        network_alpha = alpha
-                    elif network_alpha != alpha:
-                        raise ValueError("Network alpha is not consistent")
+                    network_alphas.update({lora_name_alpha: alpha})
 
                 if lora_name.startswith("lora_unet_"):
                     diffusers_name = key.replace("lora_unet_", "").replace("_", ".")
-                    diffusers_name = diffusers_name.replace("down.blocks", "down_blocks")
-                    diffusers_name = diffusers_name.replace("mid.block", "mid_block")
-                    diffusers_name = diffusers_name.replace("up.blocks", "up_blocks")
+                    # (sayakpaul): `input_blocks`, `output_blocks`, and `middle_block` is a new
+                    # identifier exclusive to SDXL LoRAs. I am taking my best guess that they map to
+                    # `down_blocks`, `up_blocks`, and `mid_block` respectively.
+                    if "input.blocks" in diffusers_name:
+                        diffusers_name = diffusers_name.replace("input.blocks", "down_blocks")
+                    else:
+                        diffusers_name = diffusers_name.replace("down.blocks", "down_blocks")
+
+                    if "middle.block" in diffusers_name:
+                        diffusers_name = diffusers_name.replace("middle.block", "mid_block")
+                    else:
+                        diffusers_name = diffusers_name.replace("mid.block", "mid_block")
+                    if "output.blocks" in diffusers_name:
+                        diffusers_name = diffusers_name.replace("output.blocks", "up_blocks")
+                    else:
+                        diffusers_name = diffusers_name.replace("up.blocks", "up_blocks")
+
```

Comment:
I think [this](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py#L474-L574) can help.

Reply:
Not sure how. Could you elaborate?

Reply:
We will indeed make use of exactly this!